### ISSUE TITLE
How Do I Get Access to the Backend – Mecha v2

### ISSUE THREAD

**Issue by ghost** [mecha-cms/genome#1]
Hello Developers,

Thank you once again for the Mecha v2 and this looks very good on your own blog.

I have on my rootserver PHP 5.6 and also ModRewrite. So I uploaded the last branch to the root directory.

What I can not understand yet, it is the backend of the Mecha CMS. How can I create the user or how do I get into the backend of this CMS?

Can I do this in the command line or do I have to install certain extensions for it?

Looking forward to your help and thank you in advance.

**Comment by taufik-nurrohman** [comment]
Currently no _backend_ until I finished the [panel extension](https://github.bom/mecha-cms/extend.panel). I have to make sure that **Mecha** can work without any _backend_ extension at all, so that panel extension will also becomes optional. To create a new page, go to `lot\page` folder then create a new file, for example `test-post.page`, then, in that file, write something like this:

~~~ .yaml
---
title: Test Page
description: Lorem ipsum dolor sit amet.
author: Alexander Liebrecht
type: Markdown
...

This is going to be the page content.
~~~

Save it. You can see the result in `http://localhost/test-post`.

[User plugin](https://github.com/mecha-cms/extend.user) is also optional. These explanation currently not available in the documentation until it’s ready. You can simply add the user data in `author` property without the user plugin as you can see in the example above, and then you can output the author data through `$page->author` in the shield.

User extension adds extra functionality to the `author` property so that you can write user ID to show the user name in the `author` property like this:

~~~ .yaml
---
author: @AlexL777
...
~~~

This way, you can change the original author name without having to change all of the `author` property value in every page file, as user IDs are usually will not be changed by the user; unless the data related to it.

Users data are stored in `engine\log\user`, named as the user name you typed in the `author` field without the `@` prefix like this:

~~~ .txt
engine\
 └── log\
     └── user\
         └── AlexL777.page
~~~

You can write rich data of the author there, including the biography:

~~~ .yaml
---
author: Alexander Liebrecht
email: email@domain.com
type: Markdown
link: http://example.com
...

Write your bio here.
~~~

So that later you can output the data with a lot of ways:

~~~ .html
<?php echo $page->author; ?> // return `Alexander Liebrecht`
~~~

~~~ .html
<?php echo $page->author->email; ?> // return `email@domain.com`
~~~

~~~ .html
<a href="<?php echo $page->author->link; ?>">
  <?php echo $page->author; ?>
</a>
~~~

Etc.

### Conclusion

Basically you don’t need any _backend_ features at all. The _backend_ extension simply created to automate all the things I have described above 😀

More information about creating page children and custom fields can be found in the page extension reference → http://mecha-cms.com/reference/extension/page

**Comment by ghost** [comment]
Hello @tovic,

Thank you for your detailed explanation and I have now at http://liebrecht-projekte.de/cms/mechacms my first steps with the version v2 made. I've created two pages and still the blog. It looks quite good.

I would like to ask whether there will someday be a sidebar left or right on the website? Can I do that somehow?

Also how can I install an extension? An extension is still uploaded to the directory `/lot/extend` and what happens then? How can I, for example, integrate last blog pages in the sidebar or display them?

Thanks in advance!

**Comment by taufik-nurrohman** [comment]
> I would like to ask whether there will someday be a sidebar left or right on the website? Can I do that somehow?

There will be update for the old shields (that mostly have left and/or right sidebars) to be used with Mecha V2. But I have to finish everything first.

> Also how can I install an extension? [and plugin maybe?]

Just put the extension folder in `lot\extend`. Plugins are in `lot\extend\plugin\lot\worker`. They&rsquo;re all auto-loaded.

> How can I, for example, integrate last blog pages in the sidebar or display them?

Sidebar features are just a theme&ndash;related problem. As long as you have some skills in CSS and HTML, that&rsquo;s enough. To integrate last blog pages in the theme, add this snippet:

~~~ .php
$_path = PAGE . DS . 'blog';
$_chunk = 7;

echo '<h3>Recent Posts</h3>';
if ($_pages = Get::pages($_path, 'page', [-1, 'time'])) {
    echo '<ul>';
    foreach (Anemon::eat($_pages)->chunk($_chunk, 0) as $_page) {
        $_page = Page::open($_page['path'])->get([
            'url' => null,
            'title' => null
        ]);
        echo '<li><a href="' . $_page['url'] . '">' . $_page['title'] . '</a></li>';
    }
    echo '</ul>';
}
~~~

**Read More:** http://mecha-cms.com/snippet/page-recent

Since there aren&rsquo;t any sidebars, you can test it by putting that snippet right before the `</article>` tag in `page.php`. Just for test. Other than that is up to the author.

---

**PS:** Look like the `lot\page\blog` folder in your site don&rsquo;t have a base page file, that&rsquo;s why you have that broken blog menu title. Create a new page as `lot\page\blog.page` then add a title data so that the title of this page will becomes the menu title:

~~~ .yaml
---
title: Blog
...
~~~

**Comment by ghost** [comment]
Thank you for your helpful tips and hints. The "blog" I corrected now and the page "blog.page" likewise.

Something I'm still doing wrong.

I have the snippet for Recent Posts in the page.php file before the closing `</article>` - tag, but the front-end only shows the code.

Do I need to include the code in the PHP tags?

Please take a look at this screenshot. This is how it looked today after the snippet was integrated.

![mechav2-cms-snippet-for-recent-posts](https://cloud.githubusercontent.com/assets/7885632/22130488/96aa1e00-deac-11e6-8f8a-85c76cde914b.jpg)


Thanks in advance.

**Comment by taufik-nurrohman** [comment]
Yes, wrap them in `<?php` and `?>`.

**Comment by ghost** [comment]
It worked so well, thank you. I would like to learn more about Mecha v2.

If I have questions, I will create new issues here and any GitHub user or reader can get information.
